pcdsdevices/targets.py

```diff
@@ -0,0 +1,167 @@
+                 x_comp=0.0, y_init=None, y_spacing=None, y_comp=0.0,
+                 name=None):
+
+        self.x_init = x_init
+        self.x_spacing = x_spacing
+        self.x_comp = x_comp
+
+        self.y_init = y_init
+        self.y_spacing = y_spacing
+        self.y_comp = y_comp
+
+        d = {'x': x, 'y': y}
+        self._stack = StageStack(d, name)
+
+        self.x = self._stack.x
+        self._xgrid = GridAxis(self.x, self.x_spacing)
+        self.y = self._stack.y
+        self._ygrid = GridAxis(self.y, self.y_spacing)
+
+        # Treat compensation axes like additional grid axes
+        if self.x_comp:
+            self._x_comp_axis = GridAxis(self.x, self.x_comp)
+
+        if self.y_comp:
+            self._y_comp_axis = GridAxis(self.y, self.y_comp)
+
+    def wm(self):
+        """
+        Return current position of X and Y axes.
+        """
+        return {'x': self.x.wm(), 'y': self.y.wm()}
+
+    def reset(self, wait=False):
+        """
+        Return to the defined initial position (x_init, y_init). Should be
+        called during experiment setup prior to using other class methods to
+        initialize the position.
+        """
+        self.x.mv(self.x_init, wait=wait)
+        self.y.mv(self.y_init, wait=wait)
+
+    def next(self, nspaces=1, wait=False):
+        """
+        Move forward (in X) by specified integer number of targets.
+        """
+        self._xgrid.advance(nspaces, 1, wait=wait)
+        if self.y_comp:
+            self._y_comp_axis.advance(nspaces, 1, wait=wait)
+
+    def back(self, nspaces=1, wait=False):
+        """
+        Move backward (in X) by specified integer number of targets.
+        """
+        self._xgrid.advance(nspaces, -1, wait=wait)
+        if self.y_comp:
+            self._y_comp_axis.advance(nspaces, -1, wait=wait)
+
+    def up(self, nspaces=1, wait=False):
+        """
+        Move to higher target position by specified integer number of targets
+        (stage moves down).
+        """
+        self._ygrid.advance(nspaces, 1, wait=wait)
+        if self.x_comp:
+            self._x_comp_axis.advance(nspaces, 1, wait=wait)
+
+    def down(self, nspaces=1, wait=False):
+        """
+        Move to lower target position by specified integer number of targets
+        (stage moves up).
+        """
+        self._ygrid.advance(nspaces, -1, wait=wait)
+        if self.x_comp:
+            self._x_comp_axis.advance(nspaces, -1, wait=wait)
+
+    def move(self, nxspaces, nyspaces, wait=False):
+        """
+        Move to a specific target position given by (xspaces, yspaces)
+        from the defined initial position. Includes compensation if defined.
+        """
+        xpos = self.x_init + self.x_spacing*nxspaces + self.x_comp*nyspaces
+        ypos = self.y_init + self.y_spacing*nyspaces + self.y_comp*nxspaces
+
+        self.x.mv(xpos, wait=wait)
+        self.y.mv(ypos, wait=wait)
```

Comment:
note that `wait=True` here will move x, then wait for x, then move y, then wait for y